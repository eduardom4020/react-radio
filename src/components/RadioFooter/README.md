# RadioFooter
>This component is a specialization of Footer component. It renders two lines with different font styles, showing the pharse 'Current Playing' and the name of the radio being played.

### Props
These are the props accepted by the component, pay attention to the description, inputs and behaviour:

---

##### radioTitle *(string)*
*The name of the radio being played*.

**example**: 
```
<RadioFooter radioTitle='Doge FM' />
```

---

##### headerStyle *(object)*
*The CSS style to apply in Header component.*

**example**: 
```
<RadioFooter headerStyle={{fontSize: 120}} />
```

---

##### titleStyle *(object)*
*The CSS style to apply in Title component.*

**example**: 
```
<RadioFooter titleStyle={{fontSize: 120}} />
```

---