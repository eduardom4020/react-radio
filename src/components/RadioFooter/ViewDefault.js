import React from 'react';
import { Text } from 'react-native';
import Footer from '~/src/components/Footer';

const RadioFooterViewDefault = props => {
    const { radioTitle } = props;

    return (
        <Footer {...props} >
            {
                radioTitle &&
                    <>
                        <Text style={{color: '#eeae61', fontSize: 12, fontWeight: 'bold'}}>CURRENTLY PLAYING</Text>
                        <Text style={{color: '#a3acbe', fontSize: 32}}>{radioTitle}</Text>
                    </>
            }
        </Footer>
    );  
};

export default RadioFooterViewDefault;