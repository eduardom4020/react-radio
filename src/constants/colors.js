export const BLACK_LIKE = '#20202b';
export const DARK_GREY = '#2e2e3a';
export const MEDIUM_GREY = '#a3acbe';
export const MEDIUM_ORANGE = '#eeae61';
export const METALIC_BLACK = '#191313';