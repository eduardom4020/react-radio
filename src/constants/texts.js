export const APP_TITLE = 'STATIONS';