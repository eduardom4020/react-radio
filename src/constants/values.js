export const DEFAULT_IMAGE_BT_SIZE = 192;
export const DEFAULT_ICON_BT_SIZE = 24;
export const DEFAULT_RADIO_PLAYING = 'CURRENTLY PLAYING'