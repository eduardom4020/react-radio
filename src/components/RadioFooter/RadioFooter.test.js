import React from 'react';
import { Text, View } from 'react-native';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { DARK_GREY } from '~/src/constants/colors';
import RadioFooter from '~/src/components/RadioFooter';

describe('RadioFooter', () => {
    test('Render?', () => {
        const component = shallow(
            <RadioFooter />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });

    test('Render with custom color?', () => {
        const component = shallow(
            <RadioFooter color={DARK_GREY} />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });

    test('Render with custom height?', () => {
        const component = shallow(
            <RadioFooter height={200} />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });
    
    test('Render with string title?', () => {
        const component = shallow(
            <RadioFooter radioTitle='Doge FM' />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });

    test('Render with component title?', () => {
        const component = shallow(
            <RadioFooter radioTitle={
                <View><Text>Doge FM</Text></View>
            } />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });
});