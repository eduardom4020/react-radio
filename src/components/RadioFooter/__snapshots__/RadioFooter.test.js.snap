// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioFooter Render with component title? 1`] = `
<Styled(View)
  radioTitle={
    <View>
      <Text>
        Doge FM
      </Text>
    </View>
  }
>
  <Styled(Text)>
    CURRENTLY PLAYING
  </Styled(Text)>
  <Styled(Text)>
    <View>
      <Text>
        Doge FM
      </Text>
    </View>
  </Styled(Text)>
</Styled(View)>
`;

exports[`RadioFooter Render with custom color? 1`] = `
<Styled(View)
  color="#2e2e3a"
/>
`;

exports[`RadioFooter Render with custom height? 1`] = `
<Styled(View)
  height={200}
/>
`;

exports[`RadioFooter Render with string title? 1`] = `
<Styled(View)
  radioTitle="Doge FM"
>
  <Styled(Text)>
    CURRENTLY PLAYING
  </Styled(Text)>
  <Styled(Text)>
    Doge FM
  </Styled(Text)>
</Styled(View)>
`;

exports[`RadioFooter Render? 1`] = `<Styled(View) />`;
